x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/notifications/SystemAuditor.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ml.notifications;
+
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.xpack.core.ml.notifications.SystemAuditMessage;
+
+public class SystemAuditor extends AbstractMlAuditor<SystemAuditMessage> {
+
+    public SystemAuditor(Client client, ClusterService clusterService) {
+        super(
+            client,
+            (resourceId, message, level, timestamp, nodeName) -> new SystemAuditMessage(message, level, timestamp, nodeName),
+            clusterService
+        );
+    }
```

Comment:
I think this should override all the audit message levels like
```
public void info(String message) {
}
public void info(String resourceId, String message) {
info(message); // Or throw unsupported?
    }
```

This way we enforce the resourceId is always passed as null and don't rely on the caller knowing this (I see mistakes in the future resulting in weird NPEs or something).